src/mongo/tree/MongoDatabaseTreeItem.ts

```diff
@@ -28,7 +28,12 @@ export class MongoDatabaseTreeItem implements IAzureParentTreeItem {
 		this._accountConnectionString = accountConnectionString;
 		this._parentId = parentId;
 		const uri: vscode.Uri = vscode.Uri.parse(accountConnectionString);
-		this.connectionString = `${uri.scheme}://${uri.authority}/${this._databaseName}?${uri.query}`;
+		const path = uri.path.substring(0, uri.path.lastIndexOf('/'));
+		if (accountConnectionString.includes('localhost')) {
+			this.connectionString = `${uri.scheme}://${uri.authority}${path}/${this._databaseName}?${uri.query}`;
+		} else {
+			this.connectionString = `${uri.scheme}://${uri.authority}/${this._databaseName}?${uri.query}`;
```

Comment:
Path only matters for localhost? It seems like we should include it in both

Something like this should work in all cases:
```
this.connectionString = `${uri.scheme}://${uri.authority}${uri.path}${this._databaseName}?${uri.query}`;
```